modules/data-table/main/index.coffee

```diff
@@ -727,19 +1064,42 @@ class DataTable extends hx.EventEmitter
   There are predefined feeds for objects and urls.
 
 ###
+whitespaceSplitRegex = /\s+/
+stripLeadingAndTrailingWhitespaceRegex = /^\s+|\s+$/g
+
+getRowSearchTerm = (cellValueLookup, row) ->
+  (v for k, v of row.cells).map(cellValueLookup).join(' ').toLowerCase()
+
+defaultTermFormatter = (term, rowSearchTerm) ->
+  arr = term?.toLowerCase()
+    .replace(stripLeadingAndTrailingWhitespaceRegex,'')
+    .split whitespaceSplitRegex
+  validPart = hx.find arr, (part) -> ~rowSearchTerm.indexOf part
+  hx.defined validPart
+
+getAdvancedSearchFilter = (cellValueLookup = hx.identity, termFormatter = defaultTermFormatter) ->
+  (filters, row) ->
+    rowSearchTerm = getRowSearchTerm(cellValueLookup, row)
+    # If term is empty this will return false
+    validFilters = hx.find filters, (groupedFilters) ->
+      invalidFilter = hx.find groupedFilters, (filter) ->
+        searchTerm = if filter.column is 'any' then rowSearchTerm else cellValueLookup(row.cells[filter.column]).toLowerCase()
+        not termFormatter filter.term, searchTerm
+      not hx.defined invalidFilter
+    hx.defined validFilters
 
 objectFeed = (data, options) ->
   options = hx.merge({
-    filter: (term, row) ->
-      rowSearchTerm = (v for k, v of row.cells).join(' ').toLowerCase() # this bit can be cached for more speedy filters
-      for part in term.toLowerCase().split(' ')
-        if rowSearchTerm.indexOf(part) is -1 then return false
-      return true
-    #XXX: should this be hx.sort.compare?
+    cellValueLookup: hx.identity
+    termFormatter: defaultTermFormatter
+
     #XXX: should this provide more information - like the column id being sorted on?
-    compare: (c1, c2) -> if c1 > c2 then 1 else -1
+    compare: hx.sort.compare
   }, options)
 
+  options.filter ?= (term, row) -> options.termFormatter(term, getRowSearchTerm(options.cellValueLookup, row))
+  options.advancedSearch ?= getAdvancedSearchFilter(options.cellValueLookup, options.termFormatter)
```

Comment:
I've updated the object feed to be more configurable - mainly because it could be difficult for people to define a custom advanced search function.
Additions are the cellValueLookup to allow cell values to be non-strings and termLookup to allow the term to be split or formatted in whatever way is required